### ISSUE TITLE
Add standard "collection" mechanism to client

### ISSUE THREAD

**Issue by forslund** [MycroftAI/mycroft-messagebus-client#20]
In the common play skills, common query skills a collection mechanism is used to request responses from multiple skills and then sift through the responses and select one. This is currently done in the following steps:

**->** A query is sent
**<-** All parties listening to the type of query (identified by message type) replies that they're intending to reply and requests a timeout to do so

**The listener notes which is trying to respond.**

**<-** listening parties sends the responses

**Responses are collected until either all searching parties has reported their answer or the timeout is reached.**

**At this point the result of the query is returned to the caller as a list / iterable.**

In short the collect method would send out a request, collect several responses and return those to the caller.

**Comment by JarbasAl** [comment]
in the better-cps framework i use 2 timeouts

- `min_timeout` -> will always wait at least this time for answers (min time ensured skills have to answer, any skill can extend it)
- `max_timeout` -> after this time, regardless of timeout extensions or number of answers, stop waiting

in practice it's just 1 timeout that skills can influence, the max_timeout is just an upper limit 

i dont/cant account for all searching parties reporting back, because there is no `do you want to answer -> gather replies` step

the aim of enforcing a `min_timeout` is to handle the scenario where a shitty skill answers really fast and would be selected when it shouldn't, the default `min_timeout` is a low value, but if a skill knows it can answer it can extend it up to `max_timeout`. This is a good compromise in regards to the inherent latency of this process
